tests/devices/i10/test_i10Apple2.py

```diff
@@ -0,0 +1,368 @@
+
+    return mock_id_pol
+
+
+@pytest.mark.parametrize(
+    "pol, top_outer_phase,top_inner_phase,btm_inner_phase, btm_outer_phase",
+    [
+        ("lh", 0, 0, 0, 0),
+        ("lv", 24.0, 0, 24.0, 0),
+        ("pc", 12, 0, 12, 0),
+        ("nc", -12, 0, -12, 0),
+        ("la", 12, 0, -12, 0),
+        ("la", 0, 12, 0, -12),
+        ("la", -11, 0, 11, 0),
+        (None, 8, 12, 2, -12),
+        (None, 11, 0, 10, 0),
+    ],
+)
+async def test_I10Apple2_determine_pol(
+    mock_id: I10Apple2,
+    pol: None | str,
+    top_inner_phase: float,
+    top_outer_phase: float,
+    btm_inner_phase: float,
+    btm_outer_phase: float,
+):
+    set_mock_value(mock_id.phase.top_inner.user_setpoint_readback, top_inner_phase)
+    set_mock_value(mock_id.phase.top_outer.user_setpoint_readback, top_outer_phase)
+    set_mock_value(mock_id.phase.btm_inner.user_setpoint_readback, btm_inner_phase)
+    set_mock_value(mock_id.phase.btm_outer.user_setpoint_readback, btm_outer_phase)
+
+    if pol is None:
+        with pytest.raises(ValueError):
+            await mock_id.set(800)
+    else:
+        await mock_id.set(800)
+
+    assert mock_id.pol == pol
+
+
+async def test_fail_I10Apple2_no_lookup(
+    mock_phaseAxes: UndlatorPhaseAxes, mock_id_gap: UndulatorGap
+):
+    wrong_path = Path("fnslkfndlsnf")
+    with pytest.raises(FileNotFoundError) as e:
+        I10Apple2(
+            id_gap=mock_id_gap,
+            id_phase=mock_phaseAxes,
+            energy_gap_table_path=wrong_path,
+            energy_phase_table_path=Path(IDPHASELOOKUPTABLE),
+            source=("Source", "idu"),
+        )
+    assert str(e.value) == f"Gap look up table is not in path: {wrong_path}"
+
+
+@pytest.mark.parametrize("energy", [(100), (2500), (-299)])
+async def test_fail_I10Apple2_set_outside_energy_limits(
+    mock_id: I10Apple2, energy: float
+):
+    with pytest.raises(ValueError) as e:
+        await mock_id.set(energy)
+    assert str(e.value) == "Demanding energy must lie between {} and {} eV!".format(
+        mock_id.lookup_tables["Gap"][mock_id.pol]["Limit"]["Minimum"],
+        mock_id.lookup_tables["Gap"][mock_id.pol]["Limit"]["Maximum"],
+    )
+
+
+async def test_fail_I10Apple2_set_lookup_gap_pol(mock_id: I10Apple2):
+    # make gap in energy
+    mock_id.lookup_tables["Gap"]["lh"]["Energies"] = {
+        "1": {
+            "Low": 255.3,
+            "High": 500,
+            "Poly": poly1d([4.33435e-08, -7.52562e-05, 6.41791e-02, 3.88755e00]),
+        }
+    }
+    mock_id.lookup_tables["Gap"]["lh"]["Energies"] = {
+        "2": {
+            "Low": 600,
+            "High": 1000,
+            "Poly": poly1d([4.33435e-08, -7.52562e-05, 6.41791e-02, 3.88755e00]),
+        }
+    }
+    with pytest.raises(ValueError) as e:
+        await mock_id.set(555)
+    assert (
+        str(e.value)
+        == "Cannot find polynomial coefficients for your requested energy."
+        + "There might be gap in the calibration lookup table."
```

Comment:
use `"""` to have multiline strings 
https://www.w3schools.com/python/gloss_python_multi_line_strings.asp